docs/source/overview/model-inference.rst

```diff
@@ -0,0 +1,77 @@
+.. _model-inference:
+
+Model Inference
+===============
+
+In EvaDB, every model is a function. We can compose SQL queries using functions as building units similar to conventional SQL functions. EvaDB's `cascades optimizer <https://faculty.cc.gatech.edu/~jarulraj/courses/8803-s21/slides/22-cascades.pdf>` will optimize the evaluation of user-defined functions for lower latency. Go over :ref:`optimizations` for more details.
+
+.. note::
+
+   EvaDB ships with a variety of builtin user-defined functions. Go over :ref:`models` to check them. Did not find the desired model? Go over :ref:`udf` to create your own user-defined functions and contribute to EvaDB.
+
+1. Projection
+
+   The most common usecases are model inference in projections. For example, we can use the `MnistImageClassifier <https://github.com/georgia-tech-db/evadb/blob/staging/evadb/functions/mnist_image_classifier.py>`_ to identify numbers from the `MINST <https://www.dropbox.com/s/yxljxz6zxoqu54v/mnist.mp4>`_ video. 
+
+.. code-block:: sql
+
+   SELECT MnistImageClassifier(data).label FROM minst_vid;
+
+2. Selection
+
+   Another common usecases are model inference in selections. In the below example, we use ``TextSummarizer`` and ``TextClassifier`` from :ref:`HuggingFace<hf>` to summarize the negative food reviews.
+
+.. code-block:: sql
+
+   SELECT TextSummarizer(data)
+   FROM food_reviews
+   WHERE TextClassifier(data).label = 'NEGATIVE';
+
+EvaDB also provides specialized array operators to construct queries. Go over built-in utility operators and functions for all of them. Below is an example of ``CONTAIN``:
+
+.. code-block:: sql
+
+   SELECT id FROM camera_videos 
+   WHERE ObjectDetector(data).labels @> ['person', 'car'];
+
+3. Lateral Join
+
+   In EvaDB, we can also use models in joins.
+   The most powerful usecase is lateral join combined with ``UNNEST``, which is very helpful to flatten the output from `one-to-many` models.
+   The key idea here is a model could give multiple outputs (e.g., bounding box) stored in an array. This syntax is used to unroll elements from the array into multiple rows.
+   Typical examples are `face detectors <https://github.com/georgia-tech-db/evadb/blob/staging/evadb/functions/face_detector.py>`_ and `object detectors <https://github.com/georgia-tech-db/evadb/blob/staging/evadb/functions/fastrcnn_object_detector.py>`_. 
+   In the below example, we use `emotion detector <https://github.com/georgia-tech-db/evadb/blob/staging/evadb/functions/emotion_detector.py>_` to detect emotions from faces in the movie, where a single scene can contain multiple faces. 
+   
+.. code-block:: sql
+   
+   SELECT EmotionDetector(Crop(data, bbox))
+   FROM movie
+   LATERAL JOIN UNNEST(FaceDetector(data)) AS Face(bbox, conf);
+
+4. Aggregate Functions
+
+   Models or user-defined functions can also act as aggregate functions for ``GROUP BY``. Typical usecases are action recognition. 
```

Comment:
We can consider changing it to something along these lines. "models can also be executed on a sequence of frames, particularly for action detection. This can be accomplished by utilizing 'GROUP BY' and 'SEGMENT' to concatenate consecutive frames into a single segment."

Reply:
We can also add the `PDF` example. @jarulraj, I remember you adding that usecase.

Reply:
Do we have a PDF example with `GROUP BY` and `SEGMENT`?

Reply:
https://github.com/georgia-tech-db/evadb/blob/04c539afade4f041a6c80ae6a340a869b525809f/test/integration_tests/long/test_huggingface_functions.py#L412